### ISSUE TITLE
Trackpy.link search range and memory

### ISSUE THREAD

**Issue by DaphneEPFL** [soft-matter/trackpy#727]
Hello, I am using TrackPy and I was wondering for the linking parameters if the search range increases with a higher memory.  For example when the memory is 2, will the search range double after a missing frame? 

In my case, I have a maximum displacement of about 10 pixels per frame and a memory of 200 frames. For my data I needed to set my search range higher than 10 to prevent the particle trajectory to be split into multiple trajectories. I would like to set my search range as low as possible to not have trajectories switch between different particles. It would help me to know if the memory parameter influences this search range.

Thank you in advance for your help!

**Comment by vivarose** [comment]
No, it doesn't expand the search range. I understand why you would want to, though! It would be a nice enhancement but I think it would be difficult to implement.

If you have flow, you might use predictors.

**Comment by vivarose** [comment]
I'm thinking about this because I also would be interested in having a maximum speed rather than a fixed search_range that doesn't adjust for memory.

Here's the relevant part of the code (in `subnet.py`).
```
    def compute(self):
        """ Evaluate the linking candidates and corresponding subnets, using
        given `search_range`."""
        source_hash = self.source_hash
        dest_hash = self.dest_hash
        if len(source_hash.points) == 0 or len(dest_hash.points) == 0:
            return
        dists, inds = source_hash.query(dest_hash.coords_mapped,
                                        self.max_neighbors, rescale=False,
                                        search_range=self.search_range)
        nn = np.sum(np.isfinite(dists), 1)  # Number of neighbors of each particle
        for i, p in enumerate(dest_hash.points):
            for j in range(nn[i]):
                wp = source_hash.points[inds[i, j]]
                wp.forward_cands.append((p, dists[i, j]))
                assign_subnet(wp, p, self.subnets)
```

In order to allow the option to vary the search_range for particles that are using memory, we would need to call source_hash.query with different search_range inputs for particles that have used memory.
